ingester/src/test_util.rs

```diff
@@ -542,3 +554,336 @@ pub async fn create_batches_with_influxtype_same_columns_different_type() -> Vec
 
     batches
 }
+
+///
+pub const TEST_NAMESPACE: &str = "test_namespace";
+///
+pub const TEST_NAMESPACE_EMPTY: &str = "test_namespace_empty";
+///
+pub const TEST_TABLE: &str = "test_table";
+///
+pub const TEST_TABLE_EMPTY: &str = "test_table_empty";
+///
+pub const TEST_PARTITION_1: &str = "test+partition_1";
+///
+pub const TEST_PARTITION_2: &str = "test+partition_2";
+
+/// Make the same in-memory data but data are split between:
+///    . one or two partition
+///    . The first partittion will have a choice to have data in either
+///       . buffer only
+///       . snapshot only
+///       . persisting only
+///       . buffer + snapshot
+///       . buffer + persisting
+///       . snpashot + persisting
+///       . buffer + snapshot + persisting
+///    . If the second partittion exists, it only has data in its buffer
+#[allow(missing_docs)]
+#[derive(Debug, PartialEq, Copy, Clone)]
+pub enum DataLocation {
+    Buffer = 1,
+    Snapshot = 2,
+    Persisting = 3,
+    BufferSnapshot = 4,
+    BufferPersisting = 5,
+    SnapshotPersisting = 6,
+    BufferSnapshotPersisting = 7,
+}
+
+/// This function produces one scenario but with the parameter combination (2*7),
+/// you will be able to produce 14 scenarios by calling it in 2 loops
+pub fn make_ingester_data(two_partitions: bool, loc: DataLocation) -> IngesterData {
+    // Whatever data because they won't be used in the tests
+    let catalog: Arc<dyn Catalog> = Arc::new(MemCatalog::new());
+    let object_store = Arc::new(ObjectStore::new_in_memory());
+    let exec = Executor::new(1);
+
+    // Make data for one sequencer/shard and two tables
+    let seq_id = SequencerId::new(1);
+    let empty_table_id = TableId::new(1);
+    let data_table_id = TableId::new(2);
+
+    // Make partitions per requested
+    let partitions = make_partitions(two_partitions, loc, seq_id, data_table_id, TEST_TABLE);
+
+    // Two tables: one empty and one with data of one or two partitions
+    let mut tables: BTreeMap<String, Arc<TableData>> = BTreeMap::default();
```

Comment:
No need for the type annotation (here and other places)

```suggestion
    let mut tables = BTreeMap::default();
```

I usually use `BTreeMap::new()` when I care to specifically make it a `BTreeMap` or I'll use `Default::default()` when I don't care and just want a fresh value.

Reply:
I will use `BTreeMap::new()` then